.github/workflows/generate-site.yaml

```diff
@@ -1,32 +1,48 @@
 name: "Generate Static Site"
+defaults:
+  run:
+    shell: bash
 on:
   workflow_call:
     inputs:
       repos:
-        required: true
+        required: false
+        type: string
+      owner:
+        required: false
+        type: string
+      name:
+        required: false
         type: string
       slug:
         required: true
         type: string
       email:
         required: true
         type: string
+      publish:
+        required: false
+        type: boolean
+        default: true
     secrets:
       id:
         required: true
       key:
         required: true
 
+env:
+  OWNER: ${{ inputs.owner || fromJSON(inputs.repos)[0].owner }}
```

Comment:
this pattern allows me to support the old and busted syntax (e.g. the one in the covidhub dashboard), to give time to transition.